### PR TITLE
Modals Update

### LINKED ISSUE #19: Properly use portals for modals
Look at https://github.com/tajo/react-portal. I'd like our modals to close on esc and click outside.

### PR DESCRIPTION
**Pull request by ydlamba**
**Modals Update**

- Close modal on _esc_ and click outside added using react-portal (PortalWithState).
- Fixes #19.

### CODE REVIEW

**Comment by ydlamba** [comment]
@domoritz This is my fix for #19. I've updated Portal to PortalWithState to add the feature for close on _esc_ and on outside click. Can you have a look?

**Comment by domoritz** [comment]
This is fantastic! I just want to ask you for one more thing. When an example is selected, the modal should close. Can you add that back?

**Comment by ydlamba** [comment]
@domoritz I've just made few changes. It's working for me. Can you check?

**Comment by domoritz** [comment]
Could you rebase to the latest changes in master? Sorry for changing so much while you were working on this.

File: src/components/header/renderer.tsx [hunk 1]
```diff
@@ -129,83 +109,122 @@ class Header extends React.Component<Props, State> {
...
+      return (
+        <div className='vega'>
+          {Object.keys(VEGA_SPECS).map((specType, i) => {
+            const specs = VEGA_SPECS[specType];
+            return (
+              <div className='itemGroup' key={i}>
+                <div className='specType'>{specType}</div>
+                <div className='items'>
+                  {specs.map((spec, j) => {
+                    return (
```
Comment (domoritz) [inline src/components/header/renderer.tsx]: Please use the more compressed markup form the left. It's easier to read.

Comment (ydlamba) [reply src/components/header/renderer.tsx]: What do you mean by compressed markup?

Comment (domoritz) [reply src/components/header/renderer.tsx]: Yeah, I like it more as it was before.

Comment (domoritz) [reply src/components/header/renderer.tsx]: Thanks for changing it!

Comment (ydlamba) [reply src/components/header/renderer.tsx]: Fixed.

Comment (ydlamba) [reply src/components/header/renderer.tsx]: I've fixed the markdowns as you wanted.

**Pull request merged**